### PR TITLE
DAOS-9672 placement: ignore further pool map changes during reintegra…

### PR DESCRIPTION
**Pull request by wangdi1**
…tion

Let's ignore the further pool map changes for layout calculation
once reintegration starts, otherwise it may cause data corruption.

See this example.

1.reintegrate rank1(targets_1 --->UP), schedule reintegrate job1.
2.reintegrate rank2(targets_2 --->UP), schedule reintegrate job2.
3.job1 will do reintegration, since targets_2 are also in UP status,
so both targets 1 and 2 will be reintegrated. But only targets_1 will
be changed to UPIN after reintegration, even though targets_2 has been
reintegrated into the new location, but it still in UP status, which can
cause data corruption.

Add extra information for pool reintegration procecess.

Fix a typo in rebuild_ec test.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

File: src/placement/jump_map.c [hunk 1]
```diff
@@ -1243,13 +1247,13 @@ jump_map_obj_find_reint(struct pl_map *map, struct daos_obj_md *md,
...
-	rc = obj_layout_alloc_and_get(jmap, &jop, md, allow_status, &layout,
-				      NULL, NULL);
+	rc = obj_layout_alloc_and_get(jmap, &jop, md, allow_status, reint_ver,
+				      &layout, NULL, NULL);
 	if (rc < 0)
 		D_GOTO(out, rc);
 
 	allow_status |= PO_COMP_ST_UP;
-	rc = obj_layout_alloc_and_get(jmap, &jop, md, allow_status,
+	rc = obj_layout_alloc_and_get(jmap, &jop, md, allow_status, reint_ver,
```
Comment (liuxuezhao) [inline src/placement/jump_map.c:1256]: the reint_ver seems same as md.omd_ver? not sure why need both two parameters of md and reint_ver for jump_map_obj_find_reint() etc.? if there are same, can obj_layout_alloc_and_get just use md.omd_ver?

**Review by liuxuezhao — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]
Is there a test that reproduces the issue?

**Comment by wangdi1** [comment]
> Is there a test that reproduces the issue?

Yes, it happens during rebuild ec test, but not always, and it only happen if we schedule new reintegration job even before the previous reintegration job starts.

File: src/placement/jump_map.c (hunk 1 above)
Comment (wangdi1) [reply src/placement/jump_map.c:1256]: hmm, it looks clearer to explicitly specify the rebuild version for find_reint() version, while md are suppose to be the object md here.  Not sure if it should be put together, I will think about it. Thanks.

**Comment by wangdi1** [comment]
please recheck, just increase the timeout for rebuild ec test, since the patch fix rebuild ec tests, which might do extra reintegration here.

**Review by liuxuezhao — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]

**Pull request merged**